dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/MPHealthTest.java

```diff
@@ -0,0 +1,241 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.checkpoint.fat;
+
+import static io.openliberty.checkpoint.fat.FATSuite.getTestMethod;
+import static io.openliberty.checkpoint.fat.FATSuite.getTestMethodNameOnly;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.io.BufferedReader;
+import java.net.HttpURLConnection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.config.ServerConfiguration;
+import com.ibm.websphere.simplicity.config.Variable;
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.annotation.Server;
+import componenttest.annotation.SkipIfCheckpointNotSupported;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.custom.junit.runner.Mode.TestMode;
+import componenttest.rules.repeater.MicroProfileActions;
+import componenttest.rules.repeater.RepeatTests;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+import componenttest.topology.utils.HttpUtils;
+import io.openliberty.checkpoint.spi.CheckpointPhase;
+
+@RunWith(FATRunner.class)
+@SkipIfCheckpointNotSupported
+public class MPHealthTest extends FATServletClient {
+
+    @Server("checkpointMPHealth")
+    public static LibertyServer server;
+
+    private static final String APP_NAME = "mphealth";
+
+    private final String HEALTH_ENDPOINT = "/health";
+    private final String READY_ENDPOINT = "/health/ready";
+    private final String STARTED_ENDPOINT = "/health/started";
+    private static final String MESSAGE_LOG = "logs/messages.log";
+
+    private final int SUCCESS_RESPONSE_CODE = 200;
+    private final int FAILED_RESPONSE_CODE = 503;
+
+    public TestMethod testMethod;
+
+    @ClassRule
+    public static RepeatTests repeatTest = MicroProfileActions.repeat("checkpointMPHealth", TestMode.FULL, MicroProfileActions.MP41, MicroProfileActions.MP50);
+
+    @BeforeClass
+    public static void copyAppToDropins() throws Exception {
+        ShrinkHelper.defaultApp(server, APP_NAME, APP_NAME);
+        FATSuite.copyAppsAppToDropins(server, APP_NAME);
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        testMethod = getTestMethod(TestMethod.class, testName);
+        server.setCheckpoint(CheckpointPhase.DEPLOYMENT, true,
```

Comment:
Does this test depend on the checkpoint phase being `DEPLOYMENT`?  What happens if `APPLICATIONS` is used?

To answer my own question: I think it would mean we timeout on applications starting on the checkpoint side which we want to avoid allowing in the future.